packages/url/init.lua

```diff
@@ -147,6 +153,12 @@ function package:registerCommands ()
         secondaryPenalty = secondaryPenalty,
         worsePenalty = worsePenalty
       })
+      local brackets = SILE.settings:get("url.brackets")
+      brackets = SU.splitUtf8(brackets)
+      if #brackets == 2 then
+        result[1] = brackets[1] .. result[1]
+        result[#result] = result[#result] .. brackets[2]
+      end
```

Comment:
FWIW, another possibility, rather than using settings, would be to call a "urlstyle" hook command (name based on "foliostyle"), instead of directly calling "code" at line 162 just below. The default implementation would just could "code" as of today, but that would allow easily changing it:
- Allowing one to use other font choices (or possibly none - my own use case, which I am solving by temporarily saving and changing "code", which is awkward and cumbersome...)
- Allowing one add brackets (your own use case here)
- Or even allowing other things for other weird use cases (e.g. we could pass the url options through to this urlstyle command, and e.g. people could come with version of it that add small pictures based on a type parameter, etc.)

Reply:
Very good idea. I often want to make all my links blue for example.

Reply:
I agree a style wrapper function would probably be better for this. That would give a lot more flexibility. Even start/end brackets may not always be a single character, what if wanted to include a thin space for example? A style wrapper that by default is a passthrough sounds like the best plan to me.